src/content/blog/astro-480.mdx

```diff
@@ -0,0 +1,193 @@
+---
+title: "Astro 4.8"
+description: "Astro 4.8 is here! This release includes experimental support for Astro actions and rewriting, performance improvements, and more."
+publishDate: "May 9, 2024"
+authors:
+  - erika
+  - ema
+  - matthew
+  - nate
+  - bjorn
+  - sarah
+coverImage: "/src/content/blog/_images/astro-480/header_astro_4_8.webp"
+socialImage: "/src/content/blog/_images/astro-480/og_image_4_8.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+
+<BlogSummary>
+Astro 4.8 is out now! This release includes experimental support for Astro actions and rewriting, performance improvements, and more.
+</BlogSummary>
+
+Full release highlights include:
+
+- [**Experimental: Astro Actions**](#experimental-astro-actions)
+- [**Experimental: Rewriting support**](#experimental-rewriting-support)
+- [**Performance improvements**](#performance-improvements)
+- [**Ability to define multiple routes with the same entrypoint**](#ability-to-define-multiple-routes-with-the-same-entrypoint)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Astro Actions
+
+This release includes the first preview for a long awaited feature: Astro actions! Actions make it easy to define and call backend functions with full type-safety from your client code.
+
+To use this feature, first enable the `experimental.actions` option in your Astro config:
+
+```js
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	experimental: {
+		actions: true
+	}
+})
+```
+
+Then, define actions in the `src/actions/index.ts` file using the `defineAction` function from `astro:actions`. You can define actions that accept JSON or form requests, and the `handler` function will be called with the parsed input and the context object:
+
+```ts
+// src/actions/index.ts
+import { defineAction, z } from "astro:actions";
+
+export const server = {
+  like: defineAction({
+    // accept json
+    input: z.object({ postId: z.string() }),
+    handler: async ({ postId }, context) => {
+      // ... update likes in db
+      return likes;
+    },
+  }),
+  comment: defineAction({
```

Comment:
Might be more succinct to pick one of the examples to show rather than showing 2. You could do: `comment: defineAction({ ... })` if you just want to allude to the fact that multiple actions can be exported.

Reply:
Yes, waiting on an example from Ben here.